include/etl/iterator.h

```diff
@@ -604,7 +604,93 @@ namespace etl
     return etl::move_iterator<TIterator>(itr);
   }
 
-#endif
+  //***************************************************************************
+  // back_insert_iterator
+  //***************************************************************************
+
+  /**
+   * @brief  Turns assignment into insertion.
+   *
+   * These are output iterators, constructed from a container-of-T.
+   * Assigning a T to the iterator appends it to the container using push_back.
+   *
+   * @tparam TContainer
+   */
+  template <class TContainer>
+  class back_insert_iterator : public iterator<output_iterator_tag, void, void, void, void>
+  {
+  public:
+    /// A nested typedef for the type of whatever container you used.
+    typedef TContainer container_type;
+
+    /// The only way to create this %iterator is with a container.
+    explicit back_insert_iterator(TContainer& c)
+      : container_(addressof(c))
+    {
+    }
+
+    /**
+     * This kind of %iterator doesn't really have a @a position in the
+     * container (you can think of the position as being permanently at
+     * the end, if you like).  Assigning a value to the %iterator will
+     * always append the value to the end of the container.
+     *
+     * @param value An instance of whatever type container_type::const_reference is;
+     *              presumably a reference-to-const T for container<T>.
+     * @return This %iterator, for chained operations.
+     */
+    back_insert_iterator& operator=(const typename TContainer::value_type& value)
+    {
+      container_->push_back(value);
+      return (*this);
+    }
+
+    back_insert_iterator& operator=(typename TContainer::value_type&& value)
```

Comment:
Should be wrapped in a check for C++11 (`ETL_HAS_CPP11` or something, don't quote me on that - there's examples of this basically wherever there's a move constructor, for example)

Reply:
It is already wrapped in a `ETL_USING_CPP11`.
see Line 435: 
https://github.com/ETLCPP/etl/blob/a84cdc6058bb7157fb22e8b06cd77cc218a241e1/include/etl/iterator.h#L435

Reply:
but I will fix that to make it clearer (closing the previous `ETL_USING_CPP11`check and reopening it.

Reply:
should I wrap also the unit test ?

Reply:
should be fixed by 37bf8b5018cc1cd444125a3b4d744f13b2c2ccb0

Reply:
I also wrapped the unit test in d43098d0a22e1296eab118a9acc563aaf1ba32ed

Reply:
My bad - although, is there any reason why `back_insert_iterator` couldn't be made available to C++03? [The STL even had it pre-C++11.](https://en.cppreference.com/w/cpp/iterator/back_insert_iterator)

Reply:
@jesseli2002 the same is true for the `reverse_iterator` implementation.
Maybe someone else can answer your question.

Reply:
There's no reason that back_insert_iterator  couldn't be made available to C++03, the other iterator types are.
Just wrap any rvalue reference parameter funtions in `ETL_USING_CPP11` compilation guards.

Reply:
You should also include, for completenes, `back_inserter`, `front_insert_iterator` & `front_inserter`.

Reply:
`front_inserter` added in 5eb8e4f6762e626a69d460f6de18930aa23bb259.